src/EditorFeatures/Test2/GoToDefinition/GoToDefinitionTests.vb

```diff
@@ -104,10 +106,13 @@ Namespace Microsoft.CodeAnalysis.Editor.UnitTests.GoToDefinition
 
         Private Sub Test(workspaceDefinition As XElement, Optional expectedResult As Boolean = True)
             Test(workspaceDefinition, expectedResult,
-                Function(document As Document, cursorPosition As Integer, presenters As IEnumerable(Of Lazy(Of IStreamingFindUsagesPresenter)))
-                    Dim goToDefService = If(document.Project.Language = LanguageNames.CSharp,
-                        DirectCast(New CSharpGoToDefinitionService(presenters), IGoToDefinitionService),
-                        New VisualBasicGoToDefinitionService(presenters))
+                Function(document As Document, cursorPosition As Integer, presenters As Lazy(Of IStreamingFindUsagesPresenter))
+                    Dim goToDefService = document.Project.Solution.Workspace.Services.GetLanguageServices(document.Project.Language).GetRequiredService(Of IGoToDefinitionService)()
+                    If document.Project.Language = LanguageNames.CSharp Then
+                        Assert.IsAssignableFrom(Of CSharpGoToDefinitionService)(goToDefService)
```

Comment:
Why are these all needed? It seems like the unit tests are very much hard-coding internal implementation, but it's unclear to me the benefit.

Reply:
> Why are these all needed? It seems like the unit tests are very much hard-coding internal implementation, but it's unclear to me the benefit.

It's simply preserving the behavior of the test from what it was before.

Reply:
This change is now removed from this pull request.